compiler/damlc/daml-compiler/src/DA/Daml/Compiler/Repl.hs

```diff
@@ -419,6 +412,64 @@ runRepl importPkgs opts replClient logger ideState = do
             Left err -> liftIO $ renderError dflags err
             Right () -> pure ()
 
+    mkReplOption
+        :: (DynFlags -> [String] -> ExceptT Error ReplM ())
+        -> [String] -> ReplM ()
+    mkReplOption option args = do
+        ReplState {lineNumber} <- State.get
+        dflags <- liftIO $
+            hsc_dflags . hscEnv <$>
+            runAction ideState (use_ GhcSession $ lineFilePath lineNumber)
+        r <- runExceptT $ option dflags args
+        case r of
+            Left err -> liftIO $ renderError dflags err
+            Right () -> pure ()
+    replOptions :: [(String, [String] -> ReplM ())]
+    replOptions =
+      [ ("help", mkReplOption optHelp)
+      , ("module", mkReplOption optModule)
+      ]
+    optHelp _dflags _args = liftIO $ T.putStrLn $ T.unlines
+      [ " Commands available from the prompt:"
+      , ""
+      , "   <statement>                 evaluate/run <statement>"
+      , "   :module [+/-] <mod> ...     add or remove the modules from the import list"
+      ]
+    optModule _dflags ("-" : modules) = do
+        ReplState {imports} <- lift State.get
+        -- TODO[AH] Use a more appropriate data structure to track imports.
```

Comment:
I got curious what ghci does and it also just has the list https://gitlab.haskell.org/ghc/ghc/-/blob/master/ghc/GHCi/UI.hs#L2524. So maybe this isn’t actually worth worrying about?

Reply:
That's true, though at least it filters out redundant imports https://gitlab.haskell.org/ghc/ghc/-/blob/52685cf7c077c51e3719e3c4dd5ca8257a99c4ea/ghc/GHCi/UI.hs#L2687.

Reply:
Good catch!